compiler/rustc_lexer/src/unescape.rs

```diff
@@ -76,22 +80,20 @@ impl EscapeError {
 /// Takes a contents of a literal (without quotes) and produces a
 /// sequence of escaped characters or errors.
 /// Values are returned through invoking of the provided callback.
+/// For `Char` and `Byte` modes, the callback will be called exactly once.
 pub fn unescape_literal<F>(src: &str, mode: Mode, callback: &mut F)
 where
     F: FnMut(Range<usize>, Result<char, EscapeError>),
 {
     match mode {
-        Mode::Char | Mode::Byte => {
+        Char | Byte => {
             let mut chars = src.chars();
-            let res = unescape_char_or_byte(&mut chars, mode == Mode::Byte);
+            let res = unescape_char_or_byte(&mut chars, mode);
```

Comment:
One year ago you wrote this https://github.com/rust-lang/rust/commit/34b32b0dac9da3fad7861bdc2bad89d771172bb3 that changed this from passing the `mode` to a bool. Does you point still stand here?

Reply:
Yeah, as mentioned in the commit log, `is_byte` doesn't really make sense since C string literals were added.

Reply:
Basically, changing to `is_byte` made sense a year ago, but enough has changed about this code since then that changing back to `mode` makes sense now :)